### PR TITLE
Fix RF100 benchmark issues

### PR DESCRIPTION
**Pull request by EHxuban11**
Fixes several low-risk issues surfaced by the RF100 benchmark thread (issue #542).

- fixes stale e2e scheduler/imgsz config for YOLO9 and RF-DETR (`yoloxwarmcos` is not accepted by either trainer; RF-DETR imgsz 560 is not divisible by its block of 32)
- centralizes RF-DETR image-size validation (train/val/predict/export now fail early with nearby valid sizes, e.g. "Use 544 or 576"); the fallback block used when a wrapper hides `patch_size`/`num_windows` matches the real detection configs (16 x 2 = 32)
- aligns EC detection defaults with its no-mosaic training path (the mosaic/color/geometric knobs were never consumed by the EC trainer)
- keeps PicoDet SimOTA probability BCE outside CUDA autocast, matching the existing YOLOX/YOLOv7 SimOTA guards
- updates stale RTMDet training-status docs (training exists behind `allow_experimental=True`)

The Volta/V100 PyTorch wheel caveat from the issue is intentionally not documented here; it will be covered outside this repo.

Validation: full unit suite green locally after rebasing onto dev, including the exporter tests that failed on the previous revision.

## Code provenance

All code and documentation changes in this PR were written specifically for this PR based on the RF100 benchmark findings in issue #542 and inspection of the LibreYOLO codebase. No third-party code was copied, ported, adapted, or derived

<details><summary><h3>Greptile Summary</h3></summary>

This PR addresses several low-severity issues surfaced by the RF100 benchmark, touching RF-DETR image-size validation, PicoDet AMP safety, EC/RTMDet config correctness, and stale YAML scheduler/imgsz values.

- **RF-DETR imgsz validation centralized**: A new `libreyolo/models/rfdetr/imgsz.py` module provides `resolve_patch_window()` and `validate_imgsz()`, wired into `_preprocess`, `_get_val_preprocessor`, `export`, `val`, and `train` on `LibreRFDETR`; `trainer.py` switches from bare `getattr(self.model, ...)` to `unwrap_model()`, removing duplicated inline block-size checks.
- **PicoDet SimOTA BCE guarded from AMP**: Probability BCE is now computed inside `torch.amp.autocast(device_type, enabled=False)` with `.float()` cast on input and `.to(iou_cost.dtype)` afterward to restore the outer dtype.
- **Config/YAML corrections**: `ECConfig` augmentation defaults zeroed to match the actual pass-through training path; `RTMDetConfig` docstring updated; RF-DETR e2e imgsz corrected from 560 to 544 and scheduler from `yoloxwarmcos` to `step`; YOLO9 scheduler corrected to `linear`.
</details>

<details><summary><h3>Confidence Score: 5/5</h3></summary>

Safe to merge; all changes are narrowly scoped bug fixes with matching unit tests and no alterations to shared training logic outside the affected models.

Each fix is self-contained: the new imgsz validation module is pure Python with no side effects, the AMP guard in PicoDet loss is proven by the new autocast smoke test, the ECConfig default changes are regression-guarded, and the YAML corrections align configs with the actual validated training paths. No regressions were identified across the 12 changed files.

No files require special attention.
</details>

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| libreyolo/models/rfdetr/imgsz.py | New centralized RF-DETR image-size validation module with resolve_patch_window(), validate_imgsz(), and suggestion helpers; well-structured and handles both scalar and tuple imgsz inputs. |
| libreyolo/models/rfdetr/model.py | Adds _validate_imgsz() wrapper and hooks it into _preprocess, _get_val_preprocessor, export, val, and train; prior review threads already cover the positional-args branch in val() and hot-path validation in _preprocess(). |
| libreyolo/models/rfdetr/trainer.py | Switches from bare getattr on self.model to unwrap_model() before reading patch_size/num_windows; delegates block-size validation to centralized validate_imgsz(), removing duplicated inline logic. |
| libreyolo/models/picodet/loss.py | Wraps probability BCE in torch.amp.autocast(device_type, enabled=False) to prevent AMP from running an unsafe op, followed by .to(iou_cost.dtype) to match the outer dtype. |
| libreyolo/export/exporter.py | Adds RF-DETR block-size validation into the base exporter, consistent with the NAFNet and DEIMv2 checks already present. |
| libreyolo/training/config.py | Resets ECConfig augmentation defaults to 0.0 to match the actual D-FINE pass-through training path; updates RTMDetConfig docstring from not implemented to implemented but experimental. |
| tests/e2e/configs/rfdetr.yaml | Corrects imgsz from 560 to 544 and switches scheduler from yoloxwarmcos to step. |
| tests/unit/test_rfdetr_imgsz_override.py | Expands coverage with a _FakeModel exposing patch_size/num_windows and adds tests for all new validation entry points. |

</details>

<details><summary><h3>Flowchart</h3></summary>

<a href="#gh-light-mode-only">

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[User calls train/val/export/_preprocess] --> B{entry point}
    B -->|train| C[LibreRFDETR.train]
    B -->|val| D[LibreRFDETR.val]
    B -->|export| E[LibreRFDETR.export]
    B -->|_preprocess| F[LibreRFDETR._preprocess]
    C --> H[_validate_imgsz]
    D --> H
    E --> H
    F --> H
    H --> I[resolve_patch_window - imgsz.py]
    I --> J{attrs found on model?}
    J -->|yes| K[use model patch_size/num_windows]
    J -->|no| L[use defaults 16x4]
    K --> M[validate_imgsz - imgsz.py]
    L --> M
    M -->|invalid| N[ValueError + nearby valid sizes]
    M -->|valid| O[proceed to super call]
```

</a>
<a href="#gh-dark-mode-only">

```mermaid
%%{init: {'theme': 'base', 'themeVariables': {"darkMode": true, "background": "#0d1117", "primaryColor": "#21262d", "primaryTextColor": "#e6edf3", "primaryBorderColor": "#8b949e", "lineColor": "#8b949e", "textColor": "#e6edf3", "edgeLabelBackground": "#161b22", "actorBkg": "#21262d", "actorBorder": "#8b949e", "actorTextColor": "#e6edf3", "actorLineColor": "#8b949e", "signalColor": "#8b949e", "signalTextColor": "#e6edf3", "noteBkgColor": "#373320", "noteBorderColor": "#d4a72c", "noteTextColor": "#f0e6c0", "labelBoxBkgColor": "#21262d", "labelBoxBorderColor": "#8b949e", "labelTextColor": "#e6edf3", "loopTextColor": "#e6edf3", "activationBkgColor": "#30363d", "activationBorderColor": "#8b949e"}}}%%
flowchart TD
    A[User calls train/val/export/_preprocess] --> B{entry point}
    B -->|train| C[LibreRFDETR.train]
    B -->|val| D[LibreRFDETR.val]
    B -->|export| E[LibreRFDETR.export]
    B -->|_preprocess| F[LibreRFDETR._preprocess]
    C --> H[_validate_imgsz]
    D --> H
    E --> H
    F --> H
    H --> I[resolve_patch_window - imgsz.py]
    I --> J{attrs found on model?}
    J -->|yes| K[use model patch_size/num_windows]
    J -->|no| L[use defaults 16x4]
    K --> M[validate_imgsz - imgsz.py]
    L --> M
    M -->|invalid| N[ValueError + nearby valid sizes]
    M -->|valid| O[proceed to super call]
```

</a>
</details>

<sub>Reviews (2): Last reviewed commit: ["Fix RF100 benchmark issues"](https://github.com/libreyolo/libreyolo/commit/acfba8829301764efe1aded431315caf41eda455) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=42737640)</sub>

### CODE REVIEW

File: libreyolo/models/rfdetr/trainer.py [hunk 1]
```diff
@@ -246,19 +247,16 @@ def _ddp_static_graph(self) -> bool:
...
-            hi = lo + block_size
-            raise ValueError(
-                f"imgsz={self.config.imgsz} is not divisible by {block_size} "
-                f"(patch_size={patch_size} x num_windows={num_windows}). "
-                f"Use {lo} or {hi}."
-            )
+        validate_imgsz(
+            self.config.imgsz,
+            patch_size=patch_size,
+            num_windows=num_windows,
```
[the remark this answers is not in the document]
Comment (EHxuban11) [reply libreyolo/models/rfdetr/trainer.py]: Fixed in dc26933d: all three trainer fallback sites (detect, multi-scale, pose) now call resolve_patch_window() so the default cannot drift from the shared helper.

File: libreyolo/models/rfdetr/model.py [hunk 2]
```diff
@@ -659,6 +682,10 @@ def _preprocess(
     ) -> Tuple[torch.Tensor, Image.Image, Tuple[int, int], float]:
         """Preprocess: resize + ImageNet normalization (no letterbox)."""
         effective_res = input_size if input_size is not None else self.input_size
+        effective_res = self._validate_imgsz(
+            effective_res,
+            name="RF-DETR inference imgsz",
+        )
```
[the remark this answers is not in the document]
Comment (EHxuban11) [reply libreyolo/models/rfdetr/model.py]: Fixed in dc26933d: _preprocess now only validates a user-supplied input_size; the construction-time native size skips validation on the hot path.

File: libreyolo/models/rfdetr/model.py [hunk 3]
```diff
@@ -1050,10 +1077,27 @@ def _load_weights(self, model_path: str | dict[str, Any]):
...
+                kwargs["imgsz"],
+                name="RF-DETR validation imgsz",
+            )
+        elif len(args) >= 3 and args[2] is not None:
+            args = list(args)
+            args[2] = self._validate_imgsz(
+                args[2],
+                name="RF-DETR validation imgsz",
+            )
+            args = tuple(args)
```
[the remark this answers is not in the document]
Comment (EHxuban11) [reply libreyolo/models/rfdetr/model.py]: Fixed in dc26933d: the positional-arg branch is removed entirely. Every validator routes the effective imgsz through _get_val_preprocessor, which validates, so val() no longer inspects the base signature's parameter order.
